pkg/controllers/tunnel/tunnel.go

```diff
@@ -145,11 +145,14 @@ func (t *Controller) handleRemovedEndpoint(obj interface{}) {
 	}
 
 	klog.V(4).Infof("Informed of removed endpoint for tunnel controller object: %#v", object)
-	if err := t.ce.RemoveCable(object.Spec.CableName); err != nil {
-		utilruntime.HandleError(fmt.Errorf("error removing endpoint cable %s from engine: %v",
-			object.Spec.CableName, err))
+	myEndpoint := types.SubmarinerEndpoint{
+		Spec: object.Spec,
+	}
+	if err := t.ce.RemoveCable(myEndpoint); err != nil {
```

Comment:
... with the proposed changes described above, this becomes simply `t.ce.RemoveCable(&object.Spec)`.